ocs_ci/deployment/baremetal.py

```diff
@@ -11,3 +20,72 @@ class BAREMETALUPI:
     def __init__(self):
         logger.info("BAREMETAL UPI")
         # TODO: OCP,OCS deployment
+
+
+class BaremetalPSIUPI(Deployment):
+    """
+    All the functionalities related to BaremetalPSI- UPI deployment
+    lives here
+    """
+    def __init__(self):
+        self.cluster_name = config.ENV_DATA['cluster_name']
+        super().__init__()
+
+    class OCPDeployment(BaseOCPDeployment):
+        def __init__(self):
+            super().__init__()
+            self.flexy_instance = FlexyBaremetalPSI()
+            self.psi_conf = load_auth_config()['psi']
+            self.utils = psiutils.PSIUtils(self.psi_conf)
+
+        def deploy_prereq(self):
+            """
+            Instantiate proper flexy class here
+
+            """
+            super().deploy_prereq()
+            self.flexy_instance.deploy_prereq()
+
+        def deploy(self):
+            self.flexy_instance.deploy()
+            # Point cluster_path to cluster-dir created by flexy
+            abs_cluster_path = os.path.abspath(self.cluster_path)
+            flexy_cluster_path = os.path.join(
+                self.flexy_instance.flexy_mnt_host_dir,
+                'flexy/workdir/install-dir'
+            )
+            logger.info(
+                "Symlinking %s to %s", abs_cluster_path, flexy_cluster_path
+            )
+            os.symlink(abs_cluster_path, flexy_cluster_path)
+            self.test_cluster()
+            # We need NTP for OCS cluster to become clean
+            logger.info("creating ntp chrony")
+            run_cmd(f"oc create -f {constants.NTP_CHRONY_CONF}")
```

Comment:
do we need to wait here to reflect changes in all worker nodes, as I have seen worker nodes will be marked shedulingdisable for sometime

Reply:
@prsurve  Any suggestion on  what's the exact condition that we have to wait on ?

Reply:
yes, i have added condition check-in my pr https://github.com/red-hat-storage/ocs-ci/blob/60e344e7fa4a5ecf54d590e952ca64f26ef0959a/ocs_ci/utility/utils.py#L2378-L2392

Reply:
Reply from Pratik,

======================================================
you can check for machineconfigpools.machineconfiguration.openshift.io
def wait_for_machineconfigpool_status(resource_name, timeout=800):
    """
    Check for Machineconfigpool status

    Args:
        resource_name (str): The name of the resource to wait (master,worker)
        timeout (int): Time in seconds to wait:


    """
    from ocs_ci.ocs import ocp
    ocp_obj = ocp.OCP(kind=constants.MACHINECONFIGPOOL, resource_name=resource_name)
    machine_count = ocp_obj.get()['status']['machineCount']

    ocp_obj.wait_for_resource(condition=str(machine_count), column="READYMACHINECOUNT", timeout=timeout)
the above code will wait for all machine to come in ready state after ntp config is applied

================================================================